### PR TITLE
Inventory: Separate query_filters for devices and vms #140

### PR DESCRIPTION
**Pull request by DouglasHeriot**
Implement separate `device_query_filters` and `vm_query_filters` as described in #140 to allow for filtering on parameters specific to devices or vms, and still continue to fetch the other type.

Also started on #56 some unit tests for the inventory, specifically to test these changes.

### CODE REVIEW

**Comment by FragmentedPacket** [comment]
@DouglasHeriot Is this ready to review?

**Comment by DouglasHeriot** [comment]
Yes this one is ready for review.  
As it contains the start of unit testing the inventory, once approved I’ll be adding more tests to the other pull requests too.

File: tests/unit/inventory/test_nb_inventory.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+@pytest.fixture
+def inventory_fixture():
+    # TODO: Mock _fetch_information() to return static HTTP responses
+    inventory = InventoryModule()
+    inventory.api_endpoint = "https://netbox.test.endpoint:1234"
+    return inventory
+
+
+@pytest.fixture
```
Comment (FragmentedPacket) [inline tests/unit/inventory/test_nb_inventory.py:40]: Should we test against the actual parameters provided in the inventory file here?

**Comment by FragmentedPacket** [comment]
Just an FYI, i'll be cutting a new release pretty soon so if there is anything you want in `0.2.0` then get them submitted or true up reviews :)

**Comment by DouglasHeriot** [comment]
@FragmentedPacket how soon? I'll be trying to have this, #153 and a few of those other issues I opened ready in the next day or two.

**Comment by FragmentedPacket** [comment]
Would Sunday work for you?

**Comment by DouglasHeriot** [comment]
@FragmentedPacket Yeah I can aim to have things ready by Sunday. Thanks.

File: tests/unit/inventory/test_nb_inventory.py (hunk 1 above)
Comment (DouglasHeriot) [reply tests/unit/inventory/test_nb_inventory.py:41]: I suppose there's no reason not to. Just thought I'd make the tests simpler.
I've now changed this test fixture to get the list of device parameters.

The tests for `refresh_url` are already against the actual lists and test both device and query parameters.

**Comment by DouglasHeriot** [comment]
This PR is now ready for review. (I'm still working on #155)

**Pull request merged**